### PR TITLE
test: add more tests and sentinel errors

### LINKED ISSUE #16: Implement smoke tests
Implement simple table driven tests for each method to ensure expected behavior for valid, invalid and edge cases. 
Assert that all the return values are sane.

### LINKED ISSUE #17: Define sentinel errors
go-execution should define well-known errors.

### PR DESCRIPTION
**Pull request by facundomedica**
## Overview

The sentinel errors are kept very generic, as there are some cases in which an execution layer would return errors and in others it would not (like initial height being 0).

Closes #16
Partially closes #17 

## Summary by CodeRabbit

- **New Features**
	- Enhanced validations and error handling for blockchain setup and transaction operations, offering clearer, immediate feedback when inputs are invalid.
	- Improved transaction processing, including more robust injection and execution pathways.
	- Introduction of structured error messages for various blockchain operations.

- **Tests**
	- Expanded test coverage with comprehensive scenarios—including concurrent operations—to ensure greater system stability and reliability.

### CODE REVIEW

File: types/errors.go [hunk 1]
```diff
@@ -0,0 +1,56 @@
+package types
```
Comment (tac0turtle) [inline types/errors.go:1]: where did these come from?

**Review by tac0turtle — APPROVED** [review]

File: types/errors.go (hunk 1 above)
Comment (facundomedica) [reply types/errors.go:1]: Came up with some that might be useful to have, there are none defined anywhere else

**Review by Manav-Aggarwal — APPROVED** [review]
